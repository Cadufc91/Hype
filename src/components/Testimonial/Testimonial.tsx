import * as React from 'react';
import {
    TestimonialContainer,
    TestimonialHeader,
    ContainerTitle,
    Divider2,
    FeedbackList,
    FeedbackCards,
    FeedbackItem,
    Profile,
    CardInfo,
    CardTitle,
    CardClient,
    FeedbackText,
    Top,
    TopText,
    RatingText,
} from './style';
import { AiFillStar } from 'react-icons/ai';

export function Testimonial () {
    return(
        <TestimonialContainer>
            <TestimonialHeader>
                <ContainerTitle>Depoimentos</ContainerTitle>
                <Divider2 />
            </TestimonialHeader>
            <FeedbackList>
                <FeedbackCards>
                    <FeedbackItem>
                        <Profile />
                        <CardInfo>
                            <CardTitle>Sundar Pichai</CardTitle>
                            <CardClient>CEO & Founder do <span>Google</span></CardClient>
                        </CardInfo>
                    </FeedbackItem>
                    <FeedbackItem>
                        <Profile />
                        <CardInfo>
                            <CardTitle>Shantanu Narayen</CardTitle>
                            <CardClient>CEO do <span>Adobe</span></CardClient>
                        </CardInfo>
                    </FeedbackItem>
                    <FeedbackItem>
                        <Profile />
                        <CardInfo>
                            <CardTitle>Noah Levin</CardTitle>
                            <CardClient>Director of Design, <span>Figma</span></CardClient>
                        </CardInfo>
                    </FeedbackItem>
                    <FeedbackItem>
                        <Profile />
                        <CardInfo>
                            <CardTitle>Stewart Butterfield</CardTitle>
                            <CardClient>CEO & Founder do <span>Slack</span></CardClient>
                        </CardInfo>
                    </FeedbackItem>

                </FeedbackCards>
                <FeedbackText>
                    <Top>
                        <AiFillStar size={23} color='#c5c502'/>
                        <TopText>5.0 Avaliação por entrelas</TopText>
                    </Top>
                    <RatingText>“Se você está procurando alguém que desafie seu pensamento de Front e realmente vá a fundo no que é importante para os usuários, Carlos é o seu homem. Além disso, ele traz um nível de entusiasmo ao ofício que é energizante para todos que trabalham com ele.”</RatingText>
                </FeedbackText>
            </FeedbackList>
        </TestimonialContainer>
    )
};

export default Testimonial;