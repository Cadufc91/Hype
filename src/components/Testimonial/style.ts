import styled from "styled-components";

export const TestimonialContainer = styled.section`
    max-width: 68%;
    display: flex;
    align-items: center;
    flex-direction: column;
    margin-left: 15%;
    margin-top: 220px;
    margin-bottom: 127px;
`;

export const TestimonialHeader = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
`;

export const ContainerTitle = styled.h1`
    font-size: 40px;
    font-weight: 400;
    color: #fff;
    margin-bottom: 16px;
`;

export const Divider2 = styled.span`
    width: 80px;
    height: 3px;
    border: 1px solid #ED1CA6;
    background: #ED1CA6;
    border-radius: 3px;
    margin-bottom: 50px;
`;

export const FeedbackList = styled.div`
    display: flex;
    flex-direction: row;
`;

export const FeedbackCards = styled.div`
    height: 411px;
    margin-right: 44px;
`;

export const FeedbackItem = styled.div`
    width: 312px;
    height: 100px;
    align-items: center;
    background: #fff;
    margin-bottom: 4px;
    display: flex;
    `;
    
export const Profile = styled.img`
    width: 52px;
    margin-right: 16px;
    height: 52px;
    margin-left: 24px;
    border-radius: 50%;
    background: #A5A5A5;
`;

export const CardInfo = styled.div`
    
`;

export const CardTitle = styled.h4`
    font-size: 14px;
    font-weight: 400;
    margin-bottom: -10px;
`;

export const CardClient = styled.p`
    font-size: 14px;
    font-weight: 400;
    color: #A5A5A5;

    span{
        color: #ED1CA6;
    }
`;

export const FeedbackText = styled.div`
    height: 411px;
    background: #fff;
`;

export const Top = styled.div`
    padding: 52px 0 0 50px;
    display: flex;
    flex-direction: row;
    align-items: center;
`;

export const TopText = styled.p`
    margin-left: 11px;
    font-size: 18px;
    font-weight: 400;
`;

export const RatingText = styled.p`
    padding: 0 50px 50px 50px;
    font-size: 22px;
`;