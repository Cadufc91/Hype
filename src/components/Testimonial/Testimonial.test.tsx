import { render } from "@testing-library/react";
import  { Testimonial } from './Testimonial';

describe('Header', () => {
    it('should match snapshot', () => {
        const { container } = render(<Testimonial />)

        expect(container.firstChild).toMatchSnapshot()
    })
})