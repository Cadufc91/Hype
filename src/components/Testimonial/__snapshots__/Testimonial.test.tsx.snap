// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<section
  class="sc-bczRLJ jHAsVv"
>
  <div
    class="sc-gsnTZi ccVAOU"
  >
    <h1
      class="sc-dkzDqf AmAtf"
    >
      Depoimentos
    </h1>
    <span
      class="sc-hKMtZM gJirKR"
    />
  </div>
  <div
    class="sc-eCYdqJ bHZJHf"
  >
    <div
      class="sc-jSMfEi gpDDAF"
    >
      <div
        class="sc-gKXOVf fBLpbJ"
      >
        <img
          class="sc-iBkjds dkqgAN"
        />
        <div
          class="sc-ftvSup crjbSj"
        >
          <h4
            class="sc-papXJ hmtoov"
          >
            Sundar Pichai
          </h4>
          <p
            class="sc-jqUVSM eWDbCB"
          >
            CEO & Founder do 
            <span>
              Google
            </span>
          </p>
        </div>
      </div>
      <div
        class="sc-gKXOVf fBLpbJ"
      >
        <img
          class="sc-iBkjds dkqgAN"
        />
        <div
          class="sc-ftvSup crjbSj"
        >
          <h4
            class="sc-papXJ hmtoov"
          >
            Shantanu Narayen
          </h4>
          <p
            class="sc-jqUVSM eWDbCB"
          >
            CEO do 
            <span>
              Adobe
            </span>
          </p>
        </div>
      </div>
      <div
        class="sc-gKXOVf fBLpbJ"
      >
        <img
          class="sc-iBkjds dkqgAN"
        />
        <div
          class="sc-ftvSup crjbSj"
        >
          <h4
            class="sc-papXJ hmtoov"
          >
            Noah Levin
          </h4>
          <p
            class="sc-jqUVSM eWDbCB"
          >
            Director of Design, 
            <span>
              Figma
            </span>
          </p>
        </div>
      </div>
      <div
        class="sc-gKXOVf fBLpbJ"
      >
        <img
          class="sc-iBkjds dkqgAN"
        />
        <div
          class="sc-ftvSup crjbSj"
        >
          <h4
            class="sc-papXJ hmtoov"
          >
            Stewart Butterfield
          </h4>
          <p
            class="sc-jqUVSM eWDbCB"
          >
            CEO & Founder do 
            <span>
              Slack
            </span>
          </p>
        </div>
      </div>
    </div>
    <div
      class="sc-kDDrLX kwuUxr"
    >
      <div
        class="sc-iqcoie fUATsx"
      >
        <svg
          color="#c5c502"
          fill="currentColor"
          height="23"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(197, 197, 2);"
          viewBox="0 0 1024 1024"
          width="23"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
          />
        </svg>
        <p
          class="sc-crXcEl beAhNn"
        >
          5.0 Avaliação por entrelas
        </p>
      </div>
      <p
        class="sc-evZas brjeHb"
      >
        “Se você está procurando alguém que desafie seu pensamento de Front e realmente vá a fundo no que é importante para os usuários, Carlos é o seu homem. Além disso, ele traz um nível de entusiasmo ao ofício que é energizante para todos que trabalham com ele.”
      </p>
    </div>
  </div>
</section>
`;
